### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by LYANGEND**
**In this project , I have achieved the following milestones;**
Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:

    List all books.
    List all people.
    Create a person (teacher or student, not a plain Person).
    Create a book.
    Create a rental.
    List all rentals for a given person id.

In  main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:

    Present the user with a list of options to perform.
    Lets users choose an option.
    If needed, ask for parameters for the option.
    Have a way to quit the app.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require_relative('./person')
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    @classroom = classroom
+    super(age, name, parent_permission)
```
Comment (CollinsTatang) [inline student.rb]: Hi @LYANGEND 
Your app throws back an error when you enter the student's name. Please fix this as requested by the project requirement.
### **Take a look**
student

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class Rental
+  attr_reader :book, :person
+  attr_accessor :date
+
+  def initialize(person, book, date)
+    @date = date
+
+    @person = person
+    person.rentals.push(self)
```
Comment (CollinsTatang) [inline rental.rb]: Hi @LYANGEND
Your app throws back an error when you enter the rentals date. Please fix this as requested by the project requirement.
### **Take a look**
rental

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  def list_people
+    if @people.empty?
+      puts 'No person record created yet'
+      return
+    end
+    @people.map do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
```
Comment (CollinsTatang) [inline app.rb]: Hi @LYANGEND
Your app throws back `No person record created yet` and `No person record created yet` when you enter the value to list all books and people created after a teacher and a book were successfully created. Please fix this as requested by the project requirement.
### **Take a look**
lists

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @LYANGEND, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
Hi @LYANGEND,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**